### ISSUE TITLE
cal_spe_func_perc()

### ISSUE THREAD

**Issue by FreitasAndy** [ChiLiubio/microeco#126]
Hi.

I'm having troubles to run the functional prediction. I have four datasets built from the same database with the same variables, from the same sequencing and all processed together. When I execute:

t1 <- trans_func$new(dataset.bk)
t1$cal_spe_func(prok_database = "FAPROTAX")
t1$cal_spe_func_perc(abundance_weighted = FALSE)

I have a good result. However, when I run the same lines for a very similar dataset:

t4 <- trans_func$new(dataset.ced)
t4$cal_spe_func(prok_database = "FAPROTAX")
t4$cal_spe_func_perc(abundance_weighted = FALSE)

I receive the message:
Error in `[.data.frame`(., , apply(., 2, sum) != 0) : 
  undefined columns selected

What could be happening? Does it mean that FAPROTAX could not make the prediction for the second dataset?

Thanks for your help.

**Comment by ChiLiubio** [comment]
Hi @FreitasAndy 

Could you please send me your dataset.ced to check what is happened? Now I am not sure what is the reason. Please save the dataset.ced using save function (https://chiliubio.github.io/microeco_tutorial/notes.html#save-function) and paste the compressed format here.
Thanks.

Chi

**Comment by FreitasAndy** [comment]
First of all, thanks for your help and for the great package you've developed.

I studied the function and saw that the problem is at the end of it. When the code is:

res_spe_func_perc %<>% t %>% 
				as.data.frame %>% 
				`colnames<-`(colnames(res_spe_func)) %>% 
				.[, apply(., 2, sum) != 0]

the last part suggests an undefined columns error.

I investigated the whole cal_spe_func function and the problem is that one sample is having NA outputs during the function execution, as follows:

> head(res_spe_func_perc)
         E1    E2   E28   E29    E3 E30    E4    E5    E6    E7    E8    E9
 [1,]  0.39  0.43  0.28  0.36  0.23  NA  0.30  0.27  0.31  0.27  0.29  0.20
 [2,]  0.00  0.00  0.00  0.00  0.00  NA  0.00  0.00  0.00  0.00  0.00  0.00
 [3,]  0.00  0.00  0.00  0.00  0.00  NA  0.00  0.00  0.00  0.00  0.00  0.00
 [4,]  0.00  0.00  0.00  0.00  0.00  NA  0.00  0.00  0.00  0.00  0.00  0.00
 [5,]  0.04  0.00  0.00  0.00  0.00  NA  0.00  0.00  0.00  0.00  0.00  0.00
 [6,]  0.00  0.00  0.00  0.00  0.00  NA  0.04  0.00  0.00  0.04  0.00  0.00

I don't know what kind of problem is happening with this sample (E30). I suppose if there were no ASVs related to that process the output will be zero, as we can see in other samples. 

After days of work, I fixed it only by remaking the dataset.
